### ISSUE TITLE
Logos support

### ISSUE THREAD

**Issue by timsueberkrueb** [timsueberkrueb/plug#1]
Hey @maciejhirsz, this is an idea that I had for a while and I would love to hear your thoughts about it, if you have the time.

This whole thing obviously relies on the assumption that it is reasonable to provide a uniform interface for different lexers which may turn out to be false (?).

The interface also probably has to be adjusted, consider it as an idea/proof-of-concept.

**Comment by maciejhirsz** [comment]
This seems like a cool idea! 

I can see why you'd want to do that for YALR. It's not really clear for me whether there is much of a need for universal lexer-parser composability, but it might be a chicken-egg problem where nobody asks for it because nobody has seen it yet?

**Comment by timsueberkrueb** [comment]
Yes, that's a good point and yeah, that's hard to tell :thinking: 

I think it would be a good idea to wait until we have more experience with YALR and/or have gathered more information about other Lexer/Parser libraries and how they would or would not benefit from this and then revisit this idea.
